UnityProject/Assets/Scripts/Systems/Score/ScoreMachine.cs

```diff
@@ -0,0 +1,183 @@
+					{
+						ScoreName = scoreName,
+						Category = category,
+						Alignment = alignment
+					};
+					Instance.Scores.Add(ID, newEntryString);
+					break;
+				default:
+					throw new ArgumentOutOfRangeException(nameof(type), type, null);
+			}
+		}
+
+		/// <summary>
+		/// How much do you want to add or subtract from a score? (Use negative numbers to subtract from the score.)
+		/// </summary>
+		public static void AddToScoreInt(int valueToAddOnTop, string ID)
+		{
+			if (Instance.Scores.ContainsKey(ID) == false)
+			{
+				Logger.LogError($"[ScoreMachine] - {ID} does not exist in the score machine!");
+				return;
+			}
+
+			if (Instance.Scores[ID] is not ScoreEntryInt c)
+			{
+				Logger.LogError($"[ScoreMachine] - Attempted to add an integer to {ID} but it's entry is not a ScoreEntryInt!");
+				return;
+			}
+			c.Score += valueToAddOnTop;
+		}
+
+		/// <summary>
+		/// whats the new bool for this score?
+		/// </summary>
+		public static void AddToScoreBool(bool newValue, string ID)
+		{
+			if (Instance.Scores.ContainsKey(ID) == false)
+			{
+				Logger.LogError($"[ScoreMachine] - {ID} does not exist in the score machine!");
+				return;
+			}
+
+			if (Instance.Scores[ID] is not ScoreEntryBool c)
+			{
+				Logger.LogError($"[ScoreMachine] - Attempted to change a bool in {ID} but it's entry is not a ScoreEntryBool!");
+				return;
+			}
+			c.Score = newValue;
+		}
+
+		/// <summary>
+		/// whats the new string for this score?
+		/// </summary>
+		public static void AddToScoreString(String newValue, string ID)
+		{
+			if (Instance.Scores.ContainsKey(ID) == false)
+			{
+				Logger.LogError($"[ScoreMachine] - {ID} does not exist in the score machine!");
+				return;
+			}
+
+			if (Instance.Scores[ID] is not ScoreEntryString c)
+			{
+				Logger.LogError($"[ScoreMachine] - Attempted to change a string in {ID} but it's entry is not a ScoreEntryString!");
+				return;
+			}
+			c.Score = newValue;
+		}
+
+		/// <summary>
+		/// Returns the int score winner from a select number of entries.
+		/// </summary>
+		public string ScoreIntWinner(List<string> IDs)
+		{
+			var highestScoreIndex = -1;
+			var winner = "";
+			foreach (var id in IDs)
+			{
+				if (Scores.ContainsKey(id) == false)
+				{
+					Logger.LogError($"[ScoreMachine] - {id} does not exist in the score machine!");
+					continue;
+				}
+				if (Scores[id] is not ScoreEntryInt c || c.Score <= highestScoreIndex) continue;
+				highestScoreIndex = c.Score;
+				winner = id;
+			}
+
+			return winner;
+		}
+
+		/// <summary>
+		/// Returns an entry that has the highest int score.
+		/// </summary>
+		/// <returns></returns>
+		public string ScoreIntWinner()
+		{
+			var highestScoreIndex = -1;
+			var winner = "";
+			foreach (var id in Scores.Keys)
+			{
+				if (Scores[id] is not ScoreEntryInt c || c.Score <= highestScoreIndex) continue;
+				highestScoreIndex = c.Score;
+				winner = id;
+			}
+
+			return winner;
```

Comment:
```suggestion
			return ScoreIntWinner(Scores.Keys);
```

Reply:
This function is used for a specific list of stuff that the programmer decides, the other one is used for all scores tracked already.

Reply:
Sure, but it does the same thing: get the highest score. Why should that logic be duplicated?

Reply:
Error detetction

Reply:
You could move the loop without the check into a method and call it from both places, where the public variant first checks all of them against the dictionary.

Whatever man, if you don't want to do it, we'll just leave it.